api/src/opentrons/cli/analyze.py

```
@@ -41,15 +65,107 @@
 @click.option(
     "--json-output",
     help="Return analysis results as machine-readable JSON.",
-    type=click.Path(path_type=AsyncPath),
+    type=click.File(mode="wb"),
```

FYI, reading machine-readable output from stdout is unreliable because if we're analyzing a Python protocol, it can *also* print to stdout, and that will corrupt the JSON stream.

Should we add a `print()` warning if `-` is specified, so it will fail early and obviously if somebody tries to rely on this?

---

IMO no, but we should put a warning in the help text. Somebody will only know to use that flag by actually reading the help text, so that should be good enough.

That's a good secondary improvement too - there's no reason we can't capture stdout and save it somewhere separate.